src/StreamFactoryInterface.php

```diff
@@ -9,15 +9,11 @@
     /**
      * Create a new stream from a resource or a string.
      *
-     * If the argument is a resource, it MUST be readable and SHOULD be seekable. It MAY be writable.
-     * If the argument is a string, a temporary resource will be created that is writable and seekable.
-     * 
-     * If the arugment is a string it will be interpided as the content of the stream. File names or 
-     * file paths are not supported. 
+     * String content will be used as the body of the stream. Files paths are not supported.
```

Comment:
@weierophinney all valid arguments, I can't debate any of that.

However, your argument works against `TempStreamFactory` just the same, e.g.:

```php
$factory = new ResourceStreamFactory();
$stream = $factory->createResourceStream(fopen('php://temp', 'w+'));
```

With the understanding I have of this now, I think I'd actually recommend an even simpler, singular interface than previously proposed.

```php
interface StreamFactoryInterface
{
    /**
     * @param resource $handle
     *
     * @return StreamInterface
     */
    public function createStream($handle);
}
```

If the only type of stream we're going to support with a factory interface, is a resource stream, then there's no longer any reason to qualify it.

There's also no reason to overload it with the ability to create a temporary stream and populate it, since that's just another resource stream, and thereby irrelevant as far as interoperability goes.

We arrive again at the situation where someone can write *one* stream factory helper class that works for *all* implementations of PSR-7 supported by a single resource stream factory implementation, e.g.:

```php
class ResourceStreamFactory
{
    private $sf;

    public function __construct(StreamFactoryInterface $sf)
    {
        $this->sf = $sf;
    }

    public function openFileStream($path, $mode = "r") {
        return $this->sf->createStream(fopen($path, $mode));
    }

    public function openTempStream($content = "") {
        $handle = fopen("php://temp", "w+");
        if ($content) fwrite($handle, $content);
        return $this->sf->createStream($handle);
    }
}
```

If we have this, and e.g. a global server request factory as I [suggested](https://github.com/http-interop/http-factory/issues/18) earlier tonight, and possibly other utility classes that work with any PSR-7 implementation via implementations of *minimal* factory interface solely for interop, perhaps packaged as e.g. `http-interop/util`, I think that's actually a lot more simple and elegant than requiring everyone else to implement a bunch more interfaces and redundant code and tests.

I like where this is going. @shadowhand your feelings?